eng/pipelines/preview-typespec.yml

```diff
@@ -0,0 +1,83 @@
+parameters:
+- name: RunUnitTests
+  displayName: Run unit tests
+  type: boolean
+  default: true
+- name: RunGenerationChecks
+  displayName: Run code generation checks
+  type: boolean
+  default: true
+
+resources:
+  repositories:
+    - repository: sdk-repository
+      type: github
+      endpoint: Azure
+      name: Azure/azure-sdk-for-net
+      ref: refs/heads/main
+    - repository: azure-sdk-tools
+      type: github
+      endpoint: Azure
+      name: Azure/azure-sdk-tools
+      ref: refs/heads/main
+
+pool:
+  name: azsdk-pool-mms-win-2022-general
+  vmImage: windows-2022
+
+variables:
+- template: /eng/pipelines/preview-variables.yml
+  parameters:
+    UseTypeSpecNext: true
+    RunUnitTests: ${{ parameters.RunUnitTests }}
+    RunGenerationChecks: ${{ parameters.RunGenerationChecks }}
+
+extends:
+  template: eng/pipelines/templates/stages/archetype-autorest-preview.yml@azure-sdk-tools
+  parameters:
+    RegenerationJobCount: ${{ variables.RegenerationJobCount }}
+    BuildPrereleaseVersion: ${{ variables.BuildPrereleaseVersion }}
+    UseTypeSpecNext: ${{ variables.UseTypeSpecNext }}
+    EmitterPackageJsonPath: ${{ variables.EmitterPackageJsonPath }}
+    PublishInternal: ${{ variables.PublishInternal }}
+    Packages:
```

Comment:
My expectation is these three parameters could be read from variables too. But unfortunately complex type is not acceptable for a variable.